File: library-android/src/main/java/de/westnordost/osmnames/AssetManagerAccess.java
package de.westnordost.osmnames;

import android.content.res.AssetManager;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;

public class AssetManagerAccess implements FileAccessAdapter
{
	private final AssetManager assetManager;
	private final String basePath;

	public AssetManagerAccess(AssetManager assetManager, String basePath)
	{
		this.assetManager = assetManager;
		this.basePath = basePath;
	}

	@Override public boolean exists(String name) throws IOException
	{
		String[] files = assetManager.list(basePath);
		for (String file : files)
		{
			if(file.equals(name)) return true;
		}
		return false;
	}

	@Override public InputStream open(String name) throws IOException
	{
		return assetManager.open(basePath + File.pathSeparator + name);
	}
}

File: library/src/androidMain/kotlin/de/westnordost/osmfeatures/AssetManagerAccess.kt
package de.westnordost.osmfeatures

import android.content.res.AssetManager
import okio.Source
import okio.source
import java.io.File

class AssetManagerAccess(private val assetManager: AssetManager, private val basePath: String): FileAccessAdapter {
    override fun exists(name: String): Boolean {
        val files: Array<String> = assetManager.list(basePath) ?: return false
        return files.contains(name)
    }

    override fun open(name: String): Source {
        return assetManager.open(basePath + File.separator + name).source()
    }
}